public/javascripts/controllers.js

```diff
@@ -1,692 +1,816 @@
+      }
 
-  for(var key in visualizations){
-    $scope.visualizations.push(visualizations[key]);
-  }
+      if($scope.visualizations.length == 0)
+        $scope.visualizations.push({});
+
+      $scope.groupFind = function(item){
+        return Object.keys(fields).indexOf(item) > -1 ? "Fields" : "Metadata";
+      };
+
+      $scope.types = {
+        'chord': {
+          type: 'chord',
+          field1: 'From',
+          field2: 'To',
+          multiple: true
+        },
+        'bar': {
+          type: 'bar',
+          field1: 'X-Axis',
+          field2: 'Y-Axis',
+          extra_fields: {'SUM': 'Count'}
+        },
+        'area': {
+          type: 'area',
+          field1: 'X-Axis',
+          field2: 'Y-Axis',
+          extra_fields: {'SUM': 'Count'}
+        },
+        'graph': {
+          type: 'graph',
+          field1: 'From',
+          field2: 'To',
+          multiple: true
+        },
+        'narratives': {
+          type: 'narratives',
+          field1: 'Date',
+          field2: 'People',
+          multiple: true
+        }
+      };
+
+      function setDefaults(){
+        $scope.visualizations.forEach(function(item){
+          if(item.maxresults == undefined || item.maxresults == null || item.maxresults == ""){
+            item.maxresults = "250";
+          }
+        });
+      };
 
-  if($scope.visualizations.length == 0)
-    $scope.visualizations.push({});
-
-  $scope.groupFind = function(item){
-    return Object.keys(fields).indexOf(item) > -1 ? "Fields" : "Metadata";
-  };
-
-  $scope.types = {
-    'chord': {
-      type: 'chord',
-      field1: 'From',
-      field2: 'To',
-      multiple: true
-    },
-    'bar': {
-      type: 'bar',
-      field1: 'X-Axis',
-      field2: 'Y-Axis',
-      extra_fields: {'SUM': 'Count'}
-    },
-    'area': {
-      type: 'area',
-      field1: 'X-Axis',
-      field2: 'Y-Axis',
-      extra_fields: {'SUM': 'Count'}
-    },
-    'graph': {
-      type: 'graph',
-      field1: 'From',
-      field2: 'To',
-      multiple: true
-    },
-    'narratives': {
-      type: 'narratives',
-      field1: 'Date',
-      field2: 'People',
-      multiple: true
-    }
-  };
 
-  function setDefaults(){
-    $scope.visualizations.forEach(function(item){
-      if(item.maxresults == undefined || item.maxresults == null || item.maxresults == ""){
-        item.maxresults = "250";
+      $scope.save = function(){
+        setDefaults();
+
+        $http.post('/api/visualizations/save', {type: type, visualizations: $scope.visualizations})
+            .success(function(data){
+              if(!data.error){
+                $modalInstance.close('saved');
+              }
+            });
+      }
+    })
+
+    .controller('userOverviewCtrl', function($scope, $modal, $http){
+
+      $scope.itemsByPage = 10;
+      $scope.showNumberOfPages = 7;
+      $scope.user = {
+        isDeleted: 0
+      };
+      $scope.isMessage = false;
+      $scope.message = [];
```

Comment:
Can't this just be a `string`? If you use a `string` you can check the length of it to see if you wan't to use it or not. The `isMessage` `bool` can also be removed if you do it this way.